src/allmydata/test/cli/test_invite.py

```diff
@@ -306,23 +324,48 @@ def test_invite_wrong_client_abilities(self):
         with open(join(priv_dir, "introducer.furl"), "w") as f:
             f.write("pb://fooblam\n")
 
-        with mock.patch('allmydata.scripts.tahoe_invite.wormhole') as w:
-            fake_wh = _create_fake_wormhole([
-                json.dumps({u"abilities": {u"client-v9000": {}}}),
-            ])
-            w.create = mock.Mock(return_value=fake_wh)
+        wormhole_server, helper = memory_server()
+        options = runner.Options()
+        options.wormhole = wormhole_server
+        reactor = object()
 
-            rc, out, err = yield run_cli(
+        async def server():
+            rc, out, err = await run_cli(
                 "-d", intro_dir,
                 "invite",
                 "--shares-needed", "1",
                 "--shares-happy", "1",
                 "--shares-total", "1",
                 "foo",
+                options=options,
             )
             self.assertNotEqual(rc, 0)
             self.assertIn(u"No 'client-v1' in abilities", out + err)
 
+        async def client():
```

Comment:
It seems like the the three `client()` functions I've seen so far are almost the same except for the abilities? So maybe could be one function for less duplication.